services/src/main/java/org/exoplatform/addons/gamification/storage/cached/RuleCachedStorage.java

```diff
@@ -102,9 +102,8 @@ public List<RuleEntity> findAllChallengesByUser(int offset, int limit, List<Long
 
   @Override
   public List<RuleEntity> findAllChallengesByUserByDomain(long domainId, int offset, int limit, List<Long> ids) {
-    return (List<RuleEntity>) this.ruleFutureCache.get(new CacheKey(CHALLENGE_USER_DOMAIN_CONTEXT, ids, domainId, offset, limit),
-                                                       CHALLENGE_USER_DOMAIN_CONTEXT + domainId + offset
-                                                           + Utils.getCurrentUser());
+    CacheKey key = new CacheKey(CHALLENGE_USER_DOMAIN_CONTEXT, ids, domainId, offset, limit);
+    return (List<RuleEntity>) this.ruleFutureCache.get(key, key.hashCode());
```

Comment:
To apply this change in all other methods using `CacheKey` as well please. If we discovered a bug using this notation, it's better to fix it in all locations at the same time and avoid waiting dedicated tasks to fix it one by one.